ivy_tests/test_ivy/test_functional/test_experimental/test_core/test_linalg.py

```diff
@@ -583,3 +583,39 @@ def test_cond(
         x=x[0],
         p=x[1],
     )
+
+
+@handle_test(
+    fn_tree="functional.ivy.experimental.solve_triangular",
+    dtype_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=2,
+        max_num_dims=2,
+        min_dim_size=2,
+        max_dim_size=10,
+        num_arrays=2,
+        allow_nan=False,
+        shared_dtype=True,
+    ),
```

Comment:
hey @anurag629, small change. Please generate float values for `lower`, `transpose`, and `unit_diagonal`